### PR TITLE
Move `fetch` method from `StorageManagerActor` to `StorageHandlerActor`

### PR DESCRIPTION
**Pull request by hekaisheng**
## What do these changes do?

Make all storage API like `put`, `get`, `fetch` methods call `StorageHandlerActor` directly.

### CODE REVIEW

File: mars/services/storage/core.py [hunk 1]
```diff
@@ -554,24 +553,65 @@ async def open_writer(self,
...
-        else:
-            data_info = await self._data_manager_ref.get_data_info(
-                session_id, data_key)
-            await self._clients[StorageLevel.REMOTE].fetch(data_info.object_id)
+                    data_key: str,
+                    level: StorageLevel,
+                    address: str,
+                    band_name: str,
+                    error: str):
+        from .transfer import SenderManagerActor
```
Comment (qinxuye) [inline mars/services/storage/core.py:570]: Better to separate into multiple functions, like `fetch_local`, `fetch_remote`, `fetch_via_transfer`.

File: mars/services/storage/core.py [hunk 2]
```diff
@@ -554,24 +553,65 @@ async def open_writer(self,
...
+                    sender_ref = await mo.actor_ref(
+                        address=address, uid=SenderManagerActor.default_uid())
+                    await sender_ref.send_data(
+                        session_id, data_key, self._data_manager_ref.address, level)
+                logger.debug(f'finish fetching data {data_key} from {address}')
+                if not isinstance(data_key, tuple):
+                    # no need to update meta for shuffle data
+                    await meta_api.add_chunk_bands(
+                        data_key, [(address, band_name or 'numa-0')])
+            except (KeyError, mo.ActorNotExist):
```
Comment (qinxuye) [inline mars/services/storage/core.py:605]: A bit confused due to different situation, maybe the error can be unified?

**Review by qinxuye — COMMENTED** [review]
I left some comments.

**Review by wjsi — APPROVED** [review]
LGTM

**Review by qinxuye — APPROVED** [review]
LGTM

**Pull request merged**